src/main/java/org/subethamail/smtp/server/SessionHandler.java

```diff
@@ -0,0 +1,11 @@
+package org.subethamail.smtp.server;
+
+import org.subethamail.smtp.DropConnectionException;
+
+public interface SessionHandler {
+
+    public void acquire(Session session) throws DropConnectionException;
```

Comment:
Using Exceptions for control flow is not ideal from a performance perspective (although I see your example use of a `SessionHandler` was aware of this). All we need to know from this method is if it the session was not accepted then what was the failure message. Stack traces are irrelevant so let's ensure they are not involved.

What about renaming the method to 
```java
void Acceptance accept(Session session);
```
Primary constructor of `Acceptance` might look like this `Acceptance(boolean accepted, String rejectionMessage)`.

By the way `public` modifiers are redundant in interface methods. Leave them out.

I'll review the rest of the PR once we've resolved this issue.

Reply:
I agree with the above on the proposal of naming this accept since that is the only action it takes (accept/reject). 

It also makes sense to do away the DropConnectionException since it is expected behaviour that a session handler will not always accept a given session. If there was an unexpected problem or checked problem while actually 'accepting' the session **then** it would warrant an exception to be thrown, this is not the case here.

Reply:
I'll remove exceptional behaviour and change method names